webapp/channels/src/components/admin_console/system_users/system_users_list_actions/index.tsx

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import type {UserProfile} from '@mattermost/types/users';
+
+import {isSystemAdmin, isGuest} from 'mattermost-redux/utils/user_utils';
+
+import * as Menu from 'components/menu';
+
+interface Props {
+    tableId?: string;
+    rowIndex: number;
+    userRoles: UserProfile['roles'];
+    currentUserRoles: UserProfile['roles'];
+}
+
+function AdminConsoleListActions(props: Props) {
+    const {formatMessage} = useIntl();
+
+    function getTranslatedUserRole(userRoles: UserProfile['roles']) {
+        if (userRoles.length > 0 && isSystemAdmin(userRoles)) {
+            return formatMessage({
```

Comment:
This is more of a question for my own knowledge: is there a reason to prefer `formatMessage` the function vs `FormattedMessage` the component? I guess I can see the function if we have some data that we need to pass in that changes based on a loop or some other non-props related factor, but I would assume the component is better in most cases.

0/5 on changing it though, I was just curious.

Reply:
I tend to use function over components in places where the addition of a span tag is of no concern (which is created with the use of component). So yes in this case we could just use the component

Reply:
Okay, I'll leave it to you to make the decision on what to use. Thanks for the insight.